mesg/mesg.go

```diff
@@ -0,0 +1,34 @@
+package mesg
+
+import "github.com/mesg-foundation/core/container"
+
+// MESG gives all functionalies of MESG core.
+type MESG struct {
+	container *container.Container
+}
+
+// Option is a configuration func for MESG.
+type Option func(*MESG)
+
+// New creates a new MESG with given options.
+func New(options ...Option) (*MESG, error) {
+	m := &MESG{}
+	for _, option := range options {
+		option(m)
+	}
+	var err error
+	if m.container == nil {
```

Comment:
Avoid nested if's 

```
if m.container != nil {
  return m, nil
}
var err error
m.container, err = container.New()
if err ....
```

Reply:
thanks, in this case i kinda don't like making an early return here because we might add some more lines under these code that not related with initializing a container.